src/components/OrderDropdown/OrderDropdown.tsx

```diff
@@ -0,0 +1,80 @@
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import type { ButtonProps, MenuProps } from '@chakra-ui/react'
+import {
+  Button,
+  Flex,
+  Menu,
+  MenuButton,
+  MenuItemOption,
+  MenuList,
+  MenuOptionGroup,
+  Text,
+} from '@chakra-ui/react'
+import { useCallback, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+
+import { type OrderOption, OrderOptionsKeys } from './types'
+
+type OrderDropdownProps = {
+  value: OrderOptionsKeys
+  onClick: (arg: OrderOptionsKeys) => void
+  buttonProps?: ButtonProps
+} & Omit<MenuProps, 'children'>
```

Comment:
Use a `menuProps` key rather than constructing a type via `&` to prevent unnecessary re-renders by removing the `...menuProps` spread below.

```suggestion
type OrderDropdownProps = {
  value: OrderOptionsKeys
  onClick: (arg: OrderOptionsKeys) => void
  buttonProps?: ButtonProps
  menuProps?: Omit<MenuProps, 'children'>
}
```

and then below:

```ts
export const OrderDropdown: React.FC<OrderDropdownProps> = ({
  onClick,
  value,
  buttonProps,
  menuProps,
}) => {
```

Reply:
I totally removed the menuProps as it's not consumed anywhere for now in [1ac600a](https://github.com/shapeshift/web/pull/8006/commits/1ac600aeb95da5729ef78865c837fd0bcec0f777)